### PR TITLE
feat: Backwards compat with new TS LS interface

### PR DESCRIPTION
**Pull request by kyliau**
https://github.com/angular/angular/pull/31972 changes the Angular LS
interface to return TS data types. This PR makes the extension
compatible with both the old and new interface.

### CODE REVIEW

File: server/src/server.ts [hunk 1]
```diff
@@ -165,22 +162,51 @@ function logErrors<T>(f: () => T): T {
...
+    const {textSpan, definitions} = result as ts.DefinitionInfoAndBoundSpan;
+    if (!definitions || !definitions.length) {
+      return;
+    }
+    const [start, end] = documents.offsetsToPositions(textDocument, [
+      textSpan.start,
+      textSpan.start + textSpan.length,
+    ]);
+    const originSelectionRange = lsp.Range.create(
+      start.line - 1, start.character - 1, end.line - 1, end.character - 1);
```
Comment (andrius-pra) [inline server/src/server.ts]: @kyliau, you don't need to adjust line and character position. It already have correct position.

Comment (kyliau) [reply server/src/server.ts]: Isn't vscode 1-based whereas LSP is 0-based?

Comment (andrius-pra) [reply server/src/server.ts]: I don't know. I only tested changes from angular/angular#32098 and angular/angular#32085. I saw these bugs: 
"Go To Definition" doesn't work for html tags.
"Go To Definition" navigates to wrong line and position for html attributes.
 It work as expected when I remove " -1" from 186 and 194 lines

Comment (kyliau) [reply server/src/server.ts]: thanks, I'll double check this.

**Pull request merged**